src/components/Trade/TradeInput.tsx

```diff
@@ -49,6 +59,12 @@ export const TradeInput = ({ history }: RouterProps) => {
     state: { wallet }
   } = useWallet()
 
+  const cryptoBalance = useAppSelector(state =>
+    selectPortfolioCryptoHumanBalanceByAssetId(state, buyAsset?.currency?.caip19)
+  )
+  const enoughCryptoToSell = (): boolean =>
```

Comment:
Same thing here, selectors are reactive by nature so you shouldn't need to be using a function. I'd use a variable (with the same `hasValidBalance` probably) in the render method. Or a derived selector here but that's probably overkill